server/aesdsocket.c

```diff
@@ -0,0 +1,338 @@
+
+        
+        if(chdir("/") == -1)
+        {
+            syslog(LOG_ERR,"failed to chdir to /");
+        }
+        
+        umask(0);
+
+    }
+    
+
+    struct sockaddr_storage client_addr;
+    socklen_t addr_size = sizeof(client_addr);
+    char buffer[BUFFER_SIZE];
+    char sendBuffer[BUFFER_SIZE];
+
+    while (1) 
+    {
+        
+        openConSock = accept(sockfd, (struct sockaddr*)&client_addr, &addr_size);
+
+        if(openConSock == -1)
+        {
+            syslog(LOG_ERR,"connection accept failed");
+        }
+        
+        // Implentation is a combination of both cases of either 4 or 6!
+        //https://stackoverflow.com/questions/3060950/how-to-get-ip-address-from-sock-structure-in-c
+
+        char ipAccepted[INET6_ADDRSTRLEN]; // handle both Inet6 and 4
+        void *addrPtr;
+
+        // if INET 4 
+        if(client_addr.ss_family == AF_INET)
+        {
+            // then save the address
+            struct sockaddr_in *tempSock = (struct sockaddr_in *)&client_addr;
+            addrPtr = &(tempSock->sin_addr);
+        }
+        else
+        {
+            // save the INET6 address
+            struct sockaddr_in6 *tempSock = (struct sockaddr_in6 *)&client_addr;
+            addrPtr = &(tempSock->sin6_addr);
+        }
+
+        // translate the pointer to a ip using the INET_NTOP function
+        inet_ntop(client_addr.ss_family, addrPtr, ipAccepted, sizeof(ipAccepted));
+
+        syslog(LOG_INFO,"Accepted connection from %s",ipAccepted);
+
+
+        // open the writing file and create it if it doesn't exist!
+        writefp = open(DATA_PATH,   O_RDWR | O_CREAT |  O_APPEND, 0664);
+
+        if (writefp == -1 ) 
+        {
+             syslog(LOG_ERR,"Open write file failed");
+            return -1;
+        }
+
+        int bytes_received = 1;
+        char lastChar = ' ';
+
+        // While buffer recieved end is not '\n'
+        while (lastChar != '\n') 
+        {
+            // Recieve and then write recieved size to buffer size of data!
+            bytes_received = recv(openConSock, buffer, BUFFER_SIZE - 1, 0);
+
+            if(bytes_received == -1)
+            {
+                syslog(LOG_ERR,"Bytes not recieved");
+                break;
+            }
+
+            if(write(writefp,buffer,bytes_received) == -1)
+            {
+                syslog(LOG_ERR,"Bytes not written");
+
+            }
+            lastChar = buffer[bytes_received - 1];
+        }
+
+        if (bytes_received == -1) 
+        {
+            syslog(LOG_INFO,"recv fail");
+        }
+
+        // Reset file pointer
+        close(writefp);
+
+        // Open file to be read only
+        readfp = open(DATA_PATH,  O_RDONLY );
+        if (readfp == -1 ) 
+        {
+            syslog(LOG_INFO,"open fail on read");
+            return -1;
+        }
+
+
+        int bytes_send = 1;
+        lastChar = ' ';
+        // while sendBuffer's last char is not '\n'
+        while (lastChar != '\n') 
```

Comment:
@Nicholas-Buckley77 I'm not sure this is the correct logic... I think if you ever ended up with a series of packets whose total size was `BUFFER_SIZE` you'd exit early before sending the remaining bytes.